src/toxcore/dht/daemon_state.rs

```diff
@@ -0,0 +1,145 @@
+/*
+    Copyright (C) 2013 Tox project All Rights Reserved.
+    Copyright © 2018 Namsoo CHO <nscho66@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+/*!
+Save or load states of tox daemon.
+When toxcore start, it load states from saved file.
+Toxcore daemon saves its states to file every 10 minutes.
+*/
+
+use std::io::{Error, ErrorKind};
+
+use futures::{future, Stream, stream};
+
+use toxcore::dht::server::*;
+use toxcore::dht::packed_node::*;
+use toxcore::dht::server::client::*;
+use toxcore::state_format::old::*;
+use toxcore::binary_io::*;
+use toxcore::io_tokio::*;
+use toxcore::dht::kbucket::*;
+
+/// Save or load states of DHT close lists
+#[derive(Clone, Debug)]
+pub struct DaemonState;
+
+// close list has DhtNode, but when we access it with iter(), DhtNode is reformed to PackedNode
+const DHT_STATE_BUFFER_SIZE: usize =
+    // Bucket size
+    (
+        // PackedNode size
+        (
+            32 + // PK size
+            19   // SocketAddr maximum size
+        ) * BUCKET_DEFAULT_SIZE // num of DhtNodes per Bucket : 8
+    ) * KBUCKET_MAX_ENTRIES as usize; // 255
+
+impl DaemonState {
+    /// save DHT close list
+    pub fn save(server: &Server) -> Vec<u8> {
```

Comment:
rename to `serialize_old`, because you don't save it somewhere, you do actually serialize it. And `old` corresponds to old C format.

You may note it in the doc comment.